girder_utils/rest.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework.pagination import LimitOffsetPagination
+
+
+class BoundedLimitOffsetPagination(LimitOffsetPagination):
+    """
+    Pagination class that forces pagination with an upper bound.
+
+    DRF's built-in paginators such as LimitOffsetPagination will cause
+    your endpoints to return different structure based on whether or not
+    paging parameters were passed in the request. In most cases it's desirable
+    to force pagination on endpoints that could return large result sets.
+    Using this class has two benefits:
+
+    1. It will always return responses in their paged structure, i.e. a
+       JSON Object with "count", and a "results" Array. This is achieved by
+       providing a default limit that is used if none is passed.
+    2. It enforces a reasonable upper bound on page sizes. Without this, users could
+       circumvent the intent of pagination by simply providing a high enough limit.
+
+    The default limit will be the value of DRF's normal PAGE_SIZE setting, or
+    100 if that is not set or falsy. The upper bound on the limit is 1000.
+    """
+
+    max_limit = 1000
+
+    def get_limit(self, request) -> int:
+        return super().get_limit(request) or 100
```

Comment:
```suggestion
        return super().get_limit(request) or self.max_limit
```

I'd propose that we just bound this at the value of `max_limit`. That provides the same two benefits, without introducing another hardcoded magic value.

Separately, I'd suggest making the change https://github.com/girder/django-composed-configuration/pull/83 . That gives downstreams more explicit control to override the default page size.

For downstreams which set `setting.PAGE_SIZE`, either explicitly or via django-composed-configuration, this class will no longer provide the additional benefit 1. (that will always happen), but it will still provide the benefit 2.